astroid/nodes/node_classes.py

```diff
@@ -4719,9 +4715,4 @@ def const_factory(value):
 
 def is_from_decorator(node):
     """Return True if the given node is the child of a decorator"""
-    parent = node.parent
-    while parent is not None:
-        if isinstance(parent, Decorators):
-            return True
-        parent = parent.parent
-    return False
+    return any(isinstance(ancestor, Decorators) for ancestor in node.node_ancestors())
```

Comment:
Technically this is correct. However I would recommend to use a normal `for` loop instead. That will be easier to read in this case.

```suggestion
    for parent in node.ancestors():
        if isinstance(parent, Decorators):
            return True
    return False
```

Reply:
I think `any`/`all` are preferable, but there's no need to pursue that here. Actually, that gives me an idea for a new lint rule :laughing: